### PR TITLE
feat(meta): support create subscription

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Changes in meta to support create subscription.  

Follow-up PR: frontend changes in #14831

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/model_v2/src/subscription.rs
Comment (yezizp2012) [inline src/meta/model_v2/src/subscription.rs:23]: We need to define table `subscription` as well in migration files for model v2. We have defined two as you can see under the directory https://github.com/risingwavelabs/risingwave/blob/8323b17ff5e3e66e9f79eeba99971622c5908471/src/meta/model_v2/migration . These are some scripts for SQL operations, which will be automatically called during cluster upgrade. For example, like creating newly defined tables and schema changes.

Since the sql backend is used by standalone mode and it's gonna release. I suggest to generator a new one by following the readme doc and add the definition of table `subscription`.

File: proto/catalog.proto
Comment (yezizp2012) [inline proto/catalog.proto:170]: Are these two fields `db_name` and `subscription_from_name` required? If not I suggest to remove them, otherwise we have to change it accordingly if the target relation name is changed.

File: proto/user.proto
Comment (yezizp2012) [inline proto/user.proto:68]: FYI, non-related to this PR, we also have to define all available acls for subscription in https://github.com/risingwavelabs/risingwave/blob/be30f20e2c14b5d12efd5b33a4ac9e537602fc37/src/common/src/acl/mod.rs and support privilege related syntax for subscription. We can support it in an separate PR after the functionality of subscription is ready.

File: src/meta/src/manager/catalog/utils.rs
Comment (yezizp2012) [inline src/meta/src/manager/catalog/utils.rs:124]: fmt.

**Review by yezizp2012 — COMMENTED** [review]
Rest generally LGTM, thx for the job. Since so many syntaxes are supported for subscription, we need enough e2e to cover them in later PRs. Cc @zwang28 do we need some extra works to support backup functions based on etcd?

**Comment by xxhZs** [comment]
> Rest generally LGTM, thx for the job. Since so many syntaxes are supported for subscription, we need enough e2e to cover them in later PRs. Cc @zwang28 do we need some extra works to support backup functions based on etcd?

yesyes,I will add e2e in part2

File: proto/catalog.proto
Comment (hzxa21) [inline proto/catalog.proto:179]: Why skip 18-21? Please be careful and self review your codes first when doing copy-paste.

File: proto/catalog.proto
Comment (hzxa21) [inline proto/catalog.proto:170]: Why skip 9-10?

File: proto/catalog.proto
Comment (hzxa21) [inline proto/catalog.proto:166]: Why skip 4?

**Review by hzxa21 — APPROVED** [review]
Generally LGTM. Please wait for @zwang28's review before merging.

**Review by zwang28 — APPROVED** [review]
Included subscription in meta backup. LGTM for this part.

File: src/meta/model_v2/src/subscription.rs
Comment (yezizp2012) [reply src/meta/model_v2/src/subscription.rs:23]: Please add a migration file to define the table `subscription` in model v2 before merging this PR. Otherwise the sql-backend will not works for subscription.

**Review by yezizp2012 — APPROVED** [review]
LGTM

File: src/meta/src/rpc/ddl_controller.rs
Comment (wenym1) [inline src/meta/src/rpc/ddl_controller.rs:1141]: Should it be ObjectType::Subscription?

**Pull request merged**
